src/frontEnd/components/WeekView.jsx

```diff
@@ -1,192 +0,0 @@
-import React, {Component} from 'react';
-import {PropTypes} from 'prop-types';
-import BigCalendar from 'react-big-calendar';
-import {Modal,Button,Form,FormControl,FormGroup,Col} from 'react-bootstrap';
-import moment from 'moment';
-
-BigCalendar.setLocalizer(
-  BigCalendar.momentLocalizer(moment
-  ));
-
-class WeekView extends Component {
-  constructor(props) {
-    super(props);
-    this.closeModal = this.closeModal.bind(this);
-    this.showModal = this.showModal.bind(this);
-    this.onDescChange = this.onDescChange.bind(this);
-    this.onTitleChange = this.onTitleChange.bind(this);
-    this.state={
-      open:false,
-      eventModal:false,
-      eventTitle:'',
-      eventDesc:'',
-      title:'',
-      desc:'',
-      startTime:'',
-      endTime:''
-    };
-  }
-
-  onTitleChange(ev){
-    this.setState({
-      title:ev.target.value
-    });
-  }
-  onDescChange(ev){
-    this.setState({
-      desc:ev.target.value
-    });
-  }
-
-  closeModal(){
-    this.setState({
-      open: false,
-      eventModal:false
-    });
-  }
-
-  showModal(){
-    this.setState({
-      open: true,
-      eventModal:true
-    });
-  }
-
-  render() {
-
-
-    var event = {
-      summary : this.state.title,
-      description : this.state.desc,
-      startDateTime : this.state.startTime,
-      endDateTime :this.state.endTime
-    };
-    return (
-      <div className="calendar-container">
-        <div className="row">
-          <img onClick={()=>this.props.onClick('HOME')} className="col-md-1" src="images/back5.png" style={{cursor: 'pointer',width: '80px',height: '50px'}} />
-          <h1 className="col-md-11">
-            {this.props.room.room_name}
-          </h1>
-        </div>
-        <Modal show={this.state.eventModal} onHide={this.closeModal} aria-labelledby="ModalHeader">
-          <Modal.Header closeButton>
-            <Modal.Title id='ModalHeader'>Event: {this.state.eventTitle}</Modal.Title>
-          </Modal.Header>
-          <Modal.Body>
-            <Form horizontal>
-              <FormGroup>
-                <Col sm={2}>
-                  Title:
-                </Col>
-                <Col sm={10}>
-                  <FormControl type="text" value={this.state.eventTitle} disabled/>
-                </Col>
-              </FormGroup>
-              <FormGroup>
-                <Col sm={2}>
-                  Description:
-                </Col>
-                <Col sm={10}>
-                  <FormControl type="text" value={this.state.eventDesc} disabled/>
-                </Col>
-              </FormGroup>
-            </Form>
-          </Modal.Body>
-        </Modal>
-        <Modal show={this.state.open} onHide={this.closeModal} aria-labelledby="ModalHeader">
-          <Modal.Header closeButton>
-            <Modal.Title id='ModalHeader'>Create Event</Modal.Title>
-          </Modal.Header>
-          <Modal.Body>
```

Comment:
This is getting really long and hard to read. Consider splitting into components.